src/Scarf/Gateway/Rule.hs

```diff
@@ -806,6 +846,67 @@ matchPixel PixelRule {..} Request {requestWai}
   | otherwise =
       pure Nothing
 
+-- | Match a catch-all rule. In its current form the only thing it does is redirect a request 
+-- to separate domain. Something that, in the past, has been done using file rules and catch-all
+-- patterns like /{+path}. While effective URI templates are not great at dealing with encoding
+-- specifics.
+matchCatchAllV1 ::
+  (MonadMatch m) =>
+  CatchAllRuleV1 ->
+  Request ->
+  m (Maybe RedirectOrProxy)
+matchCatchAllV1 CatchAllRuleV1 {..} Request {requestWai}
+  | Just targetDomain <- ruleRedirectTargetDomain = do
+      -- Construct the absolute URL to redirect to.
+      let !absoluteUrl =
+            LBS.toStrict $
+              toLazyByteString $
+                -- Just using the protocol that the request came in with 
+                -- allows for convenient testing.
+                (if Wai.isSecure requestWai then "https://" else "http://")
+                  <> Text.encodeUtf8Builder targetDomain
+                  <> (
+                       -- We would just like to use encodePath which is provided by http-types
+                       -- but that doesn't apply all the escaping out of the box. Instead, we
+                       -- are essentially inlining its implementation here with the necessary
+                       -- adjustment.
+                       let encodedSegments =
+                             foldr
+                               (\x -> ((char7 '/' <> urlEncodeBuilder True (Text.encodeUtf8 x)) <>))
```

Comment:
We've been bitten by this in the past. It's a little bit more involved than just calling some function to encode the url segments but this way we are encoding them with max. escaping now.